### PR TITLE
Remove build_type variant from ginkgo since it's already a CMakePackage

### PR DESCRIPTION
**Pull request by jrood-nrel**
I'm not sure why this variant is here, but if it needs to be there for some reason, then it needs `RelWithDebInfo` and `MinSizeRel` in the list.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Waiting for maintainers review. Ping me in a week or so if there's no reply.

**Review by tcojean — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**
